### PR TITLE
[DNM] VXFM-6538 Add multiple PCI devices to esxi host and SVMs

### PR DESCRIPTION
**Pull request by saxens7**
Handle/Add multiple PCI devices sent to host and SVMs

### CODE REVIEW

**Comment by saxens7** [comment]
@sushilrai to be merged with https://github.com/dell-asm/asm-deployer/pull/2910

File: lib/puppet/provider/vc_vm/default.rb [hunk 1]
```diff
@@ -801,30 +803,34 @@ def available_pci_passthru_device
...
-    backing = RbVmomi::VIM.VirtualPCIPassthroughDeviceBackingInfo(
-        :id => pci_id,
-        :deviceId => pci_device_id,
-        :vendorId => vendor_id,
-        :systemId => host_uuid,
-        :deviceName => ""
-    )
+    pci_passthru_devices = available_pci_passthru_device
+    return nil if !pci_passthru_devices
+    Puppet.debug("Found an active PCI passthrough device with ids, #{pci_passthru_devices.map{|pci| pci[:id]}}")
```
Comment (sushilrai) [inline lib/puppet/provider/vc_vm/default.rb]: Use printf style 
`Puppet.debug("Found an active PCI passthrough device with ids, %s %s" % [pci_passthru_devices.map{|pci| pci[:id]])`

File: lib/puppet/provider/vc_vm/default.rb [hunk 2]
```diff
@@ -801,30 +803,34 @@ def available_pci_passthru_device
...
+    pci_passthru_device_specs = []
+    for pci_passthru_device in pci_passthru_devices
+      pci_device = find_vm_host.hardware.pciDevice.find { |pci_dev| pci_dev.id ==  pci_passthru_device.id }
+      pci_id = pci_device.id
+      pci_device_id = pci_device.deviceId.to_s(16)
+      vendor_id = pci_device.vendorId
+      host_uuid = find_vm_host.esxcli.system.uuid.get
+
```
Comment (sushilrai) [inline lib/puppet/provider/vc_vm/default.rb]: Remove extra line

File: lib/puppet/provider/vc_vm/default.rb [hunk 3]
```diff
@@ -801,30 +803,34 @@ def available_pci_passthru_device
...
+      Puppet.debug("Configuring PCI passthrough on VM with device, #{pci_device}: id=#{pci_id}, device_id=#{pci_device_id}, vendor_id=#{vendor_id}, host_uuid=#{host_uuid}")
```
Comment (sushilrai) [inline lib/puppet/provider/vc_vm/default.rb]: Use printf style debug statement

File: lib/puppet/provider/vc_vm/default.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (saxens7) [reply lib/puppet/provider/vc_vm/default.rb]: Done

File: lib/puppet/provider/vc_vm/default.rb [hunk 4]
```diff
@@ -801,30 +803,34 @@ def available_pci_passthru_device
...
-    Puppet.debug("Configuring PCI passthrough on VM with device, #{pci_device}: id=#{pci_id}, device_id=#{pci_device_id}, vendor_id=#{vendor_id}, host_uuid=#{host_uuid}")
-    backing = RbVmomi::VIM.VirtualPCIPassthroughDeviceBackingInfo(
-        :id => pci_id,
-        :deviceId => pci_device_id,
-        :vendorId => vendor_id,
-        :systemId => host_uuid,
-        :deviceName => ""
-    )
+    pci_passthru_devices = available_pci_passthru_device
+    return nil if !pci_passthru_devices
```
[the remark this answers is not in the document]
Comment (saxens7) [reply lib/puppet/provider/vc_vm/default.rb]: done

File: lib/puppet/provider/vc_vm/default.rb (hunk 1 above)
Comment (saxens7) [reply lib/puppet/provider/vc_vm/default.rb]: done

File: lib/puppet/provider/vc_vm/default.rb (hunk 2 above)
Comment (saxens7) [reply lib/puppet/provider/vc_vm/default.rb]: done

File: lib/puppet/provider/vc_vm/default.rb (hunk 3 above)
Comment (saxens7) [reply lib/puppet/provider/vc_vm/default.rb]: done

**Pull request closed without merging**